packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,394 @@
+      apk.lengthSync(),
+      0,
+      showColor: false,
+    );
+    _logger.printStatus('━' * tableWidth);
+    final Directory tempApkContent = _fileSystem.systemTempDirectory.createTempSync('flutter_tools.');
+    // TODO: implement Windows.
+    String unzipOut;
+    try {
+      unzipOut = (await _processUtils.run(<String>[
+        'unzip',
+        '-o',
+        '-v',
+        apk.path,
+        '-d',
+        tempApkContent.path
+      ])).stdout;
+    } on Exception catch (e) {
+      print(e);
+    } finally {
+      // We just want the the stdout printout. We don't need the files.
+      tempApkContent.deleteSync(recursive: true);
+    }
+
+    final SymbolNode apkAnalysisRoot = _parseUnzipFile(unzipOut);
+
+    for (final SymbolNode firstLevelPath in apkAnalysisRoot.children) {
+      _printEntitySize(
+        firstLevelPath.name,
+        firstLevelPath.value,
+        1,
+      );
+      if (firstLevelPath.name == 'lib') {
+        _printLibDetails(firstLevelPath, '', aotSizeJson);
+      }
+    }
+
+    _logger.printStatus('▒' * tableWidth);
+    
+    Map<String, dynamic> apkAnalysisJson = apkAnalysisRoot.toJson();
+    
+    // TODO(peterdjlee): Add aot size for all platforms.
+    apkAnalysisJson = addAotSizeDataToJson(
+      apkAnalysisJson,
+      'lib/arm64-v8a/libapp.so (Dart AOT)'.split('/'), 
+      json.decode(aotSizeJson.readAsStringSync()) as List<dynamic>,
+    );
+
+    // Rearrange children to ensure the type key shows at the top.
+    final List<dynamic> rootChildren = apkAnalysisJson['children'] as List<dynamic>;
+    apkAnalysisJson.remove('children');
+    
+    apkAnalysisJson['type'] = 'apk';
+    apkAnalysisJson['children'] = rootChildren;
+
+    return apkAnalysisJson;
+  }
+
+  // Parse the output of unzip -v which shows the zip's contents' compressed sizes.
+  // Example output of unzip -v:
+  //  Length   Method    Size  Cmpr    Date    Time   CRC-32   Name
+  // --------  ------  ------- ---- ---------- ----- --------  ----
+  //    11708  Defl:N     2592  78% 00-00-1980 00:00 07733eef  AndroidManifest.xml
+  //     1399  Defl:N     1092  22% 00-00-1980 00:00 f53d952a  META-INF/CERT.RSA
+  //    46298  Defl:N    14530  69% 00-00-1980 00:00 17df02b8  META-INF/CERT.SF
+
+  final RegExp _parseUnzipOutput = RegExp(r'^\s*\d+\s+[\w|:]+\s+(\d+)\s+.*  (.+)$');
+
+  SymbolNode _parseUnzipFile(String unzipOut) {
+    final Map<List<String>, int> pathsToSize = <List<String>, int>{};
+
+    // Parse each path into pathsToSize so that the key is a list of
+    // path parts and the value is the size.
+    // For example:
+    // 'path/to/file' where file = 1500 => pathsToSize[['path', 'to', 'file']] = 1500
+    for (final String line in const LineSplitter().convert(unzipOut)) {
+      final RegExpMatch match = _parseUnzipOutput.firstMatch(line);
+      if (match == null) {
+        continue;
+      }
+      pathsToSize[match.group(2).split('/')] = int.parse(match.group(1));
```

Comment:
what are we grouping by 2 for? perhaps put 2 in a const that describes what the number represents